### ISSUE TITLE
could you provide code for how to pre-train "clip-ViT-B-32-multilingual-v1" model in yourself datasets

### ISSUE THREAD

**Issue by moluchase** [UKPLab/sentence-transformers#1091]
rt,
I have <other language text, image, label> pair data, I try to feed the image to the CLIP and output the image embedding(not updated), and use the 'bert-base-multilingual-uncased' model to train the two-tower model according to [this code](https://github.com/UKPLab/sentence-transformers/blob/master/examples/training/sts/training_stsbenchmark.py) in sts, but the effect seems to be very poor. Pearson often is nan, I don’t know why, it may be my training data not too good

like this:
```
epoch/batch:0/8800, loss = 0.3350865840911865, pearson = 0.7507091401786729, spearman = 0.8944271909999159, time spent:0:36
epoch/batch:0/9000, loss = 0.37060442566871643, pearson = -0.8286021225852551, spearman = -0.8944271909999159, time spent:0:36
epoch/batch:0/9200, loss = 0.39349091053009033, pearson = 0.47060605898656643, spearman = 0.4472135954999579, time spent:0:37
epoch/batch:0/9400, loss = 0.02313607558608055, pearson = nan, spearman = nan, time spent:0:38
epoch/batch:0/9600, loss = 0.21311268210411072, pearson = -0.5195703166743391, spearman = -0.7745966692414834, time spent:0:39
epoch/batch:0/9800, loss = 0.011891317553818226, pearson = nan, spearman = nan, time spent:0:40
epoch/batch:0/10000, loss = 0.018873680382966995, pearson = nan, spearman = nan, time spent:0:41
epoch/batch:0/10200, loss = 0.023975934833288193, pearson = nan, spearman = nan, time spent:0:41
epoch/batch:0/10400, loss = 0.010975204408168793, pearson = nan, spearman = nan, time spent:0:42
epoch/batch:0/10600, loss = 0.19042155146598816, pearson = 0.5340091666588794, spearman = 0.7745966692414834, time spent:0:43
epoch/batch:0/10800, loss = 0.015369567088782787, pearson = nan, spearman = nan, time spent:0:44
epoch/batch:0/11000, loss = 0.1656089574098587, pearson = 0.9359056480990152, spearman = 0.7745966692414834, time spent:0:45
epoch/batch:0/11200, loss = 0.014439709484577179, pearson = nan, spearman = nan, time spent:0:46
epoch/batch:0/11400, loss = 0.017997238785028458, pearson = nan, spearman = nan, time spent:0:46
epoch/batch:0/11600, loss = 0.37675413489341736, pearson = 0.6632766401215453, spearman = 0.4472135954999579, time spent:0:47
```

**Comment by nreimers** [comment]
Could you post some code / colab that shows your training?

**Comment by moluchase** [comment]
> Could you post some code / colab that shows your training?

part code as follows:
```python
bert_mode_path='bert-base-multilingual-uncased'

class TextImageMatchModel(tf.keras.Model):
    def __init__(self,args):
        super(TextImageMatchModel,self).__init__()
        self.bert_encoder = TFBertModel.from_pretrained(bert_mode_path)
        self.dropout = tf.keras.layers.Dropout(args.dropout_rate)
        self.text_dense = tf.keras.layers.Dense(
            args.hidden_size,
            kernel_initializer='glorot_normal',
            bias_initializer=tf.keras.initializers.constant(0.1),
            kernel_regularizer=tf.keras.regularizers.l2(regularizers_lambda),
            bias_regularizer=tf.keras.regularizers.l2(regularizers_lambda),
            name='text_emb'
        )

    def call(self,inputs,training=True):
        text_bert_output=self.bert_encoder(inputs['input_ids'],token_type_ids=inputs['token_type_ids'],attention_mask=inputs['attention_mask'])[0]
        image_emb = inputs['image_emb']
        att_mask = tf.cast(inputs['attention_mask'],tf.float32)
        sample_len = tf.math.reduce_sum(att_mask,axis=-1,keepdims=True)
        masked_emb = text_bert_output * tf.expand_dims(att_mask,axis=-1)
        text_avg_bert_output = tf.math.reduce_sum(masked_emb,axis=1)/tf.cast(sample_len,tf.float32)
        
        text_emb = self.text_dense(text_avg_bert_output)

        image_emb_norm = tf.math.l2_normalize(image_emb,axis=1)
        text_emb_norm = tf.math.l2_normalize(text_emb,axis=1)
        sim = tf.math.reduce_sum(tf.math.multiply(image_emb_norm,text_emb_norm),axis=1)
        return sim

optimizer = tf.keras.optimizers.Adam(learning_rate=args.learning_rate)
y_pred = model(features)
loss = tf.keras.losses.MSE(y_true,y_pred)

```

**Comment by nreimers** [comment]
I am sadly not too familiar with tensorflow.
